pkg/util/condition.go

```diff
@@ -25,15 +25,15 @@ import (
 )
 
 // TryAppendCondition appends the given Condition if it's not equal to the last Condition.
-func TryAppendCondition(conditions []metav1.Condition, newCondition metav1.Condition) []metav1.Condition {
+func TryAppendCondition(conditions []metav1.Condition, newCondition *metav1.Condition) []metav1.Condition {
```

Comment:
Perhaps we should check for nil whether passed accidentally or not. We could log a message.

```suggestion
func TryAppendCondition(conditions []metav1.Condition, newCondition *metav1.Condition) []metav1.Condition {
    if newCondition == nil {
        klog.Warning("TryAppendCondition called with newCondition nil")
        return conditions
    }
```